src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1024,6 +1025,7 @@ int XrdHttpReq::ProcessHTTPReq() {
     }
     case XrdHttpReq::rtGET:
     {
+        bool ret_keepalive = keepalive; // reset() clears keepalive
```

Comment:
I think it might make sense to do
```suggestion
        int retval = keepalive ? 1 : -1; // reset() clears keepalive
```
here, then only `return retval;` later on. What do you think?

Reply:
Good idea! Change pushed.